modular_zubbers/code/modules/vending/vending.dm

```diff
@@ -0,0 +1,82 @@
+//Taken from Modular Vending in zubbers
+
+/obj/machinery/vending
+	/// Additions to the `products` list  of the vending machine, modularly. Will become null after Initialize, to free up memory.
+	var/list/zubbers_products
+	/// Additions to the `product_categories` list of the vending machine, modularly. Will become null after Initialize, to free up memory.
+	var/list/zubbers_product_categories
+	/// Additions to the `premium` list  of the vending machine, modularly. Will become null after Initialize, to free up memory.
+	var/list/zubbers_premium
+	/// Additions to the `contraband` list  of the vending machine, modularly. Will become null after Initialize, to free up memory.
+	var/list/zubbers_contraband
+
+/obj/machinery/vending/Initialize(mapload)
+	if(zubbers_products)
+		// We need this, because duplicates screw up the spritesheet!
+		for(var/item_to_add in zubbers_products)
+			products[item_to_add] = zubbers_products[item_to_add]
+
+	if(zubbers_product_categories)
+		for(var/category in zubbers_product_categories)
+			var/already_exists = FALSE
+			for(var/existing_category in product_categories)
+				if(existing_category["name"] == category["name"])
+					existing_category["products"] += category["products"]
+					already_exists = TRUE
+					break
+
+			if(!already_exists)
+				product_categories += category
+
+	if(zubbers_premium)
+		// We need this, because duplicates screw up the spritesheet!
+		for(var/item_to_add in zubbers_premium)
+			premium[item_to_add] = zubbers_premium[item_to_add]
+
+	if(zubbers_contraband)
+		// We need this, because duplicates screw up the spritesheet!
+		for(var/item_to_add in zubbers_contraband)
+			contraband[item_to_add] = zubbers_contraband[item_to_add]
+
+	// Time to make clothes amounts consistent!
+	for (var/obj/item/clothing/item in products)
+		if(products[item] < MINIMUM_CLOTHING_STOCK && allow_increase(item))
+			products[item] = MINIMUM_CLOTHING_STOCK
+
+	for (var/category in product_categories)
+		for(var/obj/item/clothing/item in category["products"])
+			if(category["products"][item] < MINIMUM_CLOTHING_STOCK && allow_increase(item))
+				category["products"][item] = MINIMUM_CLOTHING_STOCK
+
+	for (var/obj/item/clothing/item in premium)
+		if(premium[item] < MINIMUM_CLOTHING_STOCK && allow_increase(item))
+			premium[item] = MINIMUM_CLOTHING_STOCK
+
+	QDEL_NULL(zubbers_products)
+	QDEL_NULL(zubbers_product_categories)
+	QDEL_NULL(zubbers_premium)
+	QDEL_NULL(zubbers_contraband)
+	return ..()
```

Comment:
You just put in all the code I was putting off for weeks. Ok you may have your masks